server/api/crud/model_monitoring/deployment.py

```diff
@@ -1188,12 +1198,6 @@ def set_credentials(
             )
         )
 
-        self._create_stream_output(
-            stream_path=secrets_dict.get(
-                mlrun.common.schemas.model_monitoring.ProjectSecretKeys.STREAM_PATH
-            )
-        )
-
```

Comment:
To avoid creating the stream twice, which was the cause of [ML-7922](https://iguazio.atlassian.net/browse/ML-7922).

I'll add a check for access that doesn't create the stream.